### ISSUE TITLE
Javascript library callback event handler and Electric

### ISSUE THREAD

**Issue by philoskim** [hyperfiddle/electric#42]
I am using Handsontable (https://handsontable.com/) Javascript library and encountered a weird Electric behavior.

What I want is as follows.

(1) A user clicks one table row.
(2) The program finds its ID column value.
(3) The program fetches the person's more detailed information from the server.

```
(ns lab.persons
  (:require [hyperfiddle.electric :as e]
            [hyperfiddle.electric-dom2 :as dom]
            #?(:cljs [debux.cs.electric :refer-macros [clog clogn dbg dbgn]]) ))

#?(:clj (use 'debux.electric))

;;; server
#?(:clj
   (def persons!
     (atom [{:person-id 1
             :age 35
             :salary 35000
             :name "Vincent Jackson"
             :sex "Male"}
            {:person-id 2
             :age 25
             :salary 25000
             :name "Michael Tom"
             :sex "Male"}
            {:person-id 3
             :age 30
             :salary 30000
             :name "Jeniffer Mary"
             :sex "Female"}])))

#?(:clj
   (defn read-person-details [person-id]
     (some #(when (= person-id (:person-id %))
              %)
           @persons!)))


;;; client
#?(:cljs (def ^:private table! (atom nil)))

#?(:cljs (def ^:private person-id! (atom nil)))
(e/def person-id+ (e/client (e/watch person-id!)))

#?(:cljs
   (defn- get-person-id-from-table [table! row]
     (.getDataAtCell @table! row 0)))

#?(:cljs
   ;; This event handler has to be a normal js handler,
   ;; because Handsontable intercepts all the event handlers occuring inside the table
   ;; and processes them in its own way.
   ;; So I set the `person-id` into the outside atom `person-id!`.
   (defn- on-cell-mouse-up [event coord td]
     (let [row (.-row coord)
           person-id (get-person-id-from-table table! row)]
       (dbg (reset! person-id! person-id))  ;; < 1 >
       )))


;;; UI
(e/defn GetPersonDetails []
  (e/client
    ;; person-id+ is triggered by the above < 1 >
    (let [person-details (e/server (read-person-details (e/client person-id+)))]
      (dbg person-details)   ;; < 2 >
      )))

#?(:cljs
   (def ^:private table-opts*
     {:data [[1 "Jackson" "Male"]
             [2 "Tom"     "Male"]
             [3 "Mary"    "Female"]]
      :colHeaders ["ID" "Name" "Sex"]
      :licenseKey "non-commercial-and-evaluation"
      :afterOnCellMouseUp on-cell-mouse-up ;; <-- the mouse-up event handler must be here
}))

;; persons table
(e/defn Persons []
  (e/client
    (dom/div
      (let [table (js/Handsontable. dom/node (clj->js table-opts*))]
        (reset! table! table)
        (GetPersonDetails.) ))))
```

I clicked the table cells from the top (ID 1 row) to the bottom (ID 3 row) sequentially.
The result is as follows

```
;;; click on the first row
;; This print is expected
dbg: (reset! person-id! person-id) =>
|   1

;; This print is not expected but redundantly printed.
dbg: person-details =>
|   nil

;; This print is expected
dbg: person-details =>
|   {:person-id 1, :age 35, :salary 35000, :name "Vincent Jackson", :sex "Male"}



;;; click on the second row
;; This print is expected
dbg: (reset! person-id! person-id) =>
|   2

;; This print is not expected but redundantly printed.
dbg: person-details =>
|   {:person-id 1, :age 35, :salary 35000, :name "Vincent Jackson", :sex "Male"}

;; This print is expected
dbg: person-details =>
|   {:person-id 2, :age 25, :salary 25000, :name "Michael Tom", :sex "Male"}


;;; click on the third row
;; This print is expected
dbg: (reset! person-id! person-id) =>
|   3

;; This print is not expected but redundantly printed.
dbg: person-details =>
|   {:person-id 2, :age 25, :salary 25000, :name "Michael Tom", :sex "Male"}

;; This print is expected
dbg: person-details =>
|   {:person-id 3, :age 30, :salary 30000, :name "Jeniffer Mary", :sex "Female"}
```

In conclusion, the above  '< 2 >' part is weirdly called twice.

Is this a bug of Electric or is there any better solution?

The following is index.html

```
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8">

    <link rel="stylesheet" type="text/css"
          href="/lib/handsontable-11.0.1/handsontable.full.min.css">
    <script src="/lib/handsontable-11.0.1/handsontable.full.min.js"></script>
  </head>
  <body>
    <noscript>You need to enable JavaScript to run this app.</noscript>
    <script type="text/javascript" src="$:hyperfiddle.client.module/main$"></script>
  </body>
</html>
```

**Comment by xificurC** [comment]
Thank you for your report.

Is there a repo I could clone to try this out locally? Just reading it I'm not sure where the issue is, or how `dbg` is implemented

**Comment by philoskim** [comment]
I cloned the Electric Clojure and pushed this demo in the following link. 

https://github.com/philoskim/electric/blob/master/src-docs/user/demo_handsontable.cljc

You can click the 'Persons' link in the user.demo-index/Demos page. 

The debux related code is eliminated in the above demo code for your convenience but you can find the source code in the following site.

https://github.com/philoskim/debux#electric

**Comment by xificurC** [comment]
changing 
```clj
(read-person-details (e/client person-id+))
```
to
```clj
(read-person-details person-id+)
```
solves your problem. We'll investigate why and let you know. In this case the `e/client` is redundant, electric knows `person-id+` is on the client and will transfer it to the server.
